### PR TITLE
docs(tracer): Fix Tracer typing hinting for Pycharm

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

Typing for Tracer seems to be broken on PyCharm, see screenshot below:

Screen Shot 2021-03-17 at 6 39 54 AM

The location of the `Returns` doc string is causing the PyCharm linter to thing that `Tracer` is  a tuple (possibly also due to the `_default_config` and `_config` before the `__init__` ? 

## Description of changes:

Changes:
* Move up `Returns` to just below the `Parameters`
* Add type hinting for `provider` 
* Add docstring for `provider`

**Checklist**

 

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by michaelbrewer** [comment]
@heitorlessa  this seems to fix the issue with PyCharm and Tracer

File: aws_lambda_powertools/tracing/tracer.py [hunk 1]
```diff
@@ -153,7 +155,7 @@ def __init__(
         self.__build_config(
             service=service, disabled=disabled, auto_patch=auto_patch, patch_modules=patch_modules, provider=provider
         )
-        self.provider = self._config["provider"]
+        self.provider: BaseProvider = self._config["provider"]
```
Comment (heitorlessa) [inline aws_lambda_powertools/tracing/tracer.py:158]: could you remove this as it has an unintended side effect for escape hatches in X-Ray? e.g. auto-completing methods that we're not providing in the BaseProvider.

**Review by heitorlessa — CHANGES_REQUESTED** [review]
tiny change and it should be good to release!

File: aws_lambda_powertools/tracing/tracer.py (hunk 1 above)
Comment (heitorlessa) [reply aws_lambda_powertools/tracing/tracer.py:158]: Backing up as it's a Pycharm thing only, not VSCode. Best to keep it, and then later I'll add non-ABC methods for the sake of completion

**Pull request merged**